scripts/cleanup_analysis_entries.py

```diff
@@ -0,0 +1,259 @@
+from collections import defaultdict
+
+from sample_metadata.apis import ProjectApi, AnalysisApi
+from sample_metadata.model.analysis_query_model import AnalysisQueryModel
+from sample_metadata.model.analysis_status import AnalysisStatus
+from sample_metadata.model.analysis_type import AnalysisType
+
+import logging
```

Comment:
You're importing logging, but also including print statements and `warnings` - could this all be done using logging instead?

You can also set logging to show the line number the printed statement is coming from, which can help when debugging ([e.g.](https://github.com/populationgenomics/automated-interpretation-pipeline/blob/main/reanalysis/interpretation_runner.py#L451))